backend/Testing/FwHeadless/MediaFileTestFixture.cs

```diff
@@ -0,0 +1,168 @@
+        await Utils.AddMemberToProject(ProjectId, this, "manager", ProjectRole.Manager);
+        await Utils.AddMemberToProject(ProjectId, this, "editor", ProjectRole.Editor);
+        // Project folder needs to exist in order for file uploads to be allowed, so trigger first sync including S/R
+        await FwHeadlessTestHelpers.TriggerSync(HttpClient, ProjectId);
+        await FwHeadlessTestHelpers.AwaitSyncFinished(HttpClient, ProjectId);
+    }
+
+    public async Task DisposeAsync()
+    {
+        // Must ensure we're logged in as admin, since some tests might have changed that
+        await LoginIfNeeded("admin");
+        await HttpClient.DeleteAsync($"api/project/{ProjectId}");
+    }
+
+    public async Task<(FileListing?, HttpResponseMessage)> ListFiles(Guid projectId, string? relativePath = null, string loginAs = "admin")
+    {
+        await LoginIfNeeded(loginAs);
+        var url = $"/api/media/list/{projectId}";
+        if (relativePath is not null)
+        {
+            var qb = new QueryBuilder { { "relativePath", relativePath } };
+            url += qb.ToString();
+        }
+        var request = new HttpRequestMessage(HttpMethod.Get, url);
+        var result = await HttpClient.SendAsync(request);
+        return (await result.Content.ReadFromJsonAsync<FileListing>(), result);
+    }
+
+    public async Task<(Guid, HttpResponseMessage)> PostFile(string localPath, string? overrideFilename = null, FileMetadata? metadata = null, string loginAs = "admin", IDictionary<string, string>? extraFields = null)
+    {
+        await LoginIfNeeded(loginAs);
+        var filename = Path.GetFileName(localPath);
+        using (var formData = new MultipartFormDataContent())
+        {
+            if (overrideFilename is not null)
+            {
+                formData.Add(new StringContent(overrideFilename), name: "filename");
+            }
+            formData.Add(new StringContent(ProjectId.ToString()), name: "projectId");
+            if (metadata is not null)
+            {
+                if (metadata.Author is not null)
+                {
+                    formData.Add(new StringContent(metadata.Author), name: "author");
+                }
+                if (metadata.License is not null)
+                {
+                    formData.Add(new StringContent(metadata.License.Value.ToString()), name: "license");
+                }
+                if (extraFields is not null)
+                {
+                    foreach (var kv in extraFields)
+                    {
+                        formData.Add(new StringContent(kv.Value), name: kv.Key);
+                    }
+                }
+            }
+            var stream = new StreamContent(File.OpenRead(localPath));
+            formData.Add(stream, name: "file", fileName: filename);
+            var request = new HttpRequestMessage(HttpMethod.Post, $"api/media/?projectId={ProjectId}");
```

Comment:
you've got projectId here and in the Form data, I think it only needs to be in one of those places

Reply:
Done.